sql/execution_pipeline.h

```diff
@@ -331,6 +346,20 @@ struct PipelineElement: public std::enable_shared_from_this<PipelineElement> {
          std::shared_ptr<SqlExpression> where = SqlExpression::TRUE,
          OrderByExpression orderBy = OrderByExpression());
     
+    /** Add a join to the pipeline that is pre-bound into the immediate outer
+        scope.
+    */
+    std::shared_ptr<PipelineElement>
+    join(std::shared_ptr<TableExpression> left,
```

Comment:
The second one is an optimization for when the expression has already been bound previously; the first one stayed for compatibility with existing code.  Note that the first forwards to the second in the implementation.